crypto/asn1/d2i_pr.c

```diff
@@ -60,18 +61,21 @@ EVP_PKEY *d2i_PrivateKey_ex(int type, EVP_PKEY **a, const unsigned char **pp,
                 goto err;
             EVP_PKEY_free(ret);
             ret = tmp;
+            ERR_pop_to_mark();
             if (EVP_PKEY_type(type) != EVP_PKEY_base_id(ret))
                 goto err;
         } else {
             ASN1err(0, ERR_R_ASN1_LIB);
             goto err;
         }
     }
+    ERR_clear_last_mark();
```

Comment:
The clear must be called only in case ERR_pop_to_mark was not called, so it cannot be called here.

Reply:
I was thinking that this would be a noop if ERR_pop_to_mark was called but I realize now there could have been other marks set previously. Thanks, I'll revert this commit and take another stab at it.